core/src/main/scala/chisel3/experimental/Defaulting.scala

```diff
@@ -0,0 +1,48 @@
+// SPDX-License-Identifier: Apache-2.0
+
+package chisel3.experimental
+
+import chisel3.{chiselTypeOf, Record, Data, Element}
+
+import scala.collection.immutable.SeqMap
+
+/** Data type for representing any type who has a default value
+  *
+  * Uses the underlying opaque type support.
+  *
+  * @note This API is experimental and subject to change
+  */
+final class Defaulting[T <: Data] private (private[chisel3] val tpe: T, val defaultValue: T) extends Record {
```

Comment:
Because these things get defined at the top-level (outside of a Module), you probably want the defaultValue to be a thunk, not a value.

Reply:
Good point. I'll change it, thanks!